tests/e2e/test_chaperone.js

```diff
@@ -4,68 +4,171 @@ const log				= require('@whi/stdlog')(path.basename( __filename ), {
-    	log.silly("From puppeteer: console.log( %s )", msg.text() );
-    });
-    
     log.info("Go to: %s", url );
     await page.goto( url, { "waitUntil": "networkidle0" } );
 
     return page;
 }
 
-function delay(t, val) {
-    return new Promise(function(resolve) {
-	setTimeout(function() {
-	    resolve(val);
-	}, t);
-    });
+class PageTestUtils {  
+	constructor( page ) {
+	this.logPageErrors			= () => page.on('pageerror', async error => {
+		if (error instanceof Error) {
+		log.silly( error.message );
+	    }
+	    else
+		log.silly( error );
+	});
+
+	this.describeJsHandleLogs	= () => page.on('console', async msg => {
+	    const args = await Promise.all(msg.args().map(arg => this.describeJsHandle( arg )))
+		  .catch(error => console.log( error.message ));
+		console.log( ...args );
+		// log.silly("From puppeteer: console.log( %s )", ...args );
+	});
+
+	this.describeJsHandle		= ( jsHandle ) => {
+	    return jsHandle.executionContext().evaluate(arg => {
+		if (arg instanceof Error)
+		    return arg.message;
+		else
+		    return arg;
+	    }, jsHandle);
+	};
+    }
 }
 
-// NOT RANDOM: this instance_prefix matches the hha_hash hard-coded in Chaperone for DEVELOP mode
-const instance_prefix				= "QmV1NgkXFwromLvyAmASN7MbgLtgUaEYkozHPGUxcHAbSL";
-const host_agent_id				= fs.readFileSync('./AGENTID', 'utf8').trim();
+// NB: The 'host_agent_id' *is not* in the holohash format as it is a holo host pubkey (as generated from the hpos-seed)
+const host_agent_id				= 'd5xbtnrazkxx8wjxqum7c77qj919pl2agrqd3j2mmxm62vd3k'
+
 log.info("Host Agent ID: %s", host_agent_id );
 
+const envoy_mode_map = {
+	production: 0,
+	develop: 1,
+}
 
-describe("Server", () => {
+const envoyOpts = {
+	mode: envoy_mode_map.develop,
+	hosted_port_number: 0,
+	hosted_app: {
+		servicelogger_id: HOSTED_APP_SERVICELOGGER_INSTALLED_APP_ID,
+		dnas: [{
+			nick: 'test-hha', // 'test-elemental-chat',
+			path: '/home/lisa/Documents/gitrepos/holo/rsm-updated/holo-envoy/dnas/holo-hosting-app.dna.gz', // '/home/lisa/Documents/gitrepos/holo/rsm-updated/holo-envoy/dnas/elemental-chat.dna.gz',
+		}]
+	}
+}
 
+const getHostAgentKey = async (serviceClient) => {
+	const appInfo = await serviceClient.appInfo({ installed_app_id: HOSTED_APP_SERVICELOGGER_INSTALLED_APP_ID  });
+	const agentPubKey = appInfo.cell_data[0][0][1];
+	return {
+		decoded: agentPubKey,
+		encoded: Codec.AgentId.encode(agentPubKey)
+	}
+}
+// Register test app in hha  (in real word scenario - will be done when provider registers app in hha):
+const registerTestAppInHha = async (hostedClient) => {
+	const hhaAppInfo = await hostedClient.appInfo({ installed_app_id: HHA_INSTALLED_APP_ID  });
+	const hhaCellId = hhaAppInfo.cell_data[0][0];
+
+	const happBundle = {
+		hosted_url:"https://testapp.com",
+		happ_alias: "test-app",
+		ui_path: "/path/to/test_app_ui.zip",
+		name: "Test App Hosted On Web",
+		dnas: [{
+			hash: "hC0k...",
+			path: envoyOpts.hosted_app.dnas[0].path,
```

Comment:
This path will be a url, not a relative path

Reply:
This is just registering the mock dna that is already downloaded. I'll manged the url case and create a test for it too though.